vars/terraform.groovy

```diff
@@ -163,7 +163,13 @@ def call(userConfig = [:]) {
             }
             if (!isBuildCauseTimer && isBuildOnProductionBranch) {
               stage('🚢 Shipping Changes') {
-                sh makeCliCmd + ' deploy'
+                try {
+                  sh makeCliCmd + ' deploy'
+                } catch(Exception e) {
+                  // If the deploy failed, keep the pod until a user catch the problem (cloud be an errored state, or many reason to keep the workspace)
+                  input message: 'An error happened while applying the terraform plan. Keeping the agent up and running. Delete the agent?'
```

Comment:
I would save the error.tfstate as artifact here too, wdyt?

Reply:
And block subsequent builds (by not allowing builds if there is another running?)

Reply:
>  I would save the error.tfstate as artifact here too, wdyt?

Nope: this files contains sensitive datas so it should be kept in the ephemeral agents to avoid any accidental exposure

Reply:
> And block subsequent builds (by not allowing builds if there is another running?)

Good idea 🤔 I wonder if we should not use a `lock()` in this case (that would be released when the user interacts). WDYT?

Reply:
Idk the details of the lock, yes maybe? What would be locked?